probe-rs/src/architecture/arm/ap/memory_ap/mod.rs

```diff
@@ -42,17 +45,30 @@ impl From<GenericAp> for MemoryAp {
     }
 }
 
+/// The unit of data that is transferred in one transfer via the DRW commands.
+///
+/// This can be configured with the CSW command.
+///
+/// ALL MCUs support `U32`. All other transfer sizes are optionally implemented.
 #[derive(Debug, Primitive, Clone, Copy, PartialEq)]
 pub enum DataSize {
+    /// 1 byte transfers are supported.
     U8 = 0b000,
+    /// 2 byte transfers are supported.
     U16 = 0b001,
+    /// 4 byte transfers are supported.
     U32 = 0b010,
+    /// 8 byte transfers are supported.
     U64 = 0b011,
+    /// 16 byte transfers are supported.
     U128 = 0b100,
+    /// 32 byte transfers are supported.
     U256 = 0b101,
 }
 
 impl DataSize {
+    /// Create a new `DataSize` from a number of bytes.
+    /// Defaults to 4 bytes.
```

Comment:
Well, the argument, cannot (sadly) be constrained to just an arbitrary number span. So it can be 128 for example, which is not a valid input, in which case it defaults to 4 bytes.
I guess I will clarify this :)